x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/FrequentItemSetsAggregationBuilder.java

```diff
@@ -0,0 +1,182 @@
+import org.elasticsearch.search.aggregations.support.MultiValuesSourceFieldConfig;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ContextParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public final class FrequentItemSetsAggregationBuilder extends AbstractAggregationBuilder<FrequentItemSetsAggregationBuilder> {
+
+    public static final String NAME = "frequent_items";
+
+    public static final double DEFAULT_MINIMUM_SUPPORT = 0.01;
+    public static final int DEFAULT_MINIMUM_SET_SIZE = 1;
+    public static final int DEFAULT_SIZE = 10;
+
+    public static final ParseField MINIMUM_SUPPORT = new ParseField("minimum_support");
+    public static final ParseField MINIMUM_SET_SIZE = new ParseField("minimum_set_size");
+    public static final ParseField FIELDS = new ParseField("fields");
+
+    public static final ConstructingObjectParser<FrequentItemSetsAggregationBuilder, String> PARSER = new ConstructingObjectParser<>(
+        NAME,
+        false,
+        (args, context) -> {
+            @SuppressWarnings("unchecked")
+            List<MultiValuesSourceFieldConfig> fields = (List<MultiValuesSourceFieldConfig>) args[0];
+            double minimumSupport = args[1] == null ? DEFAULT_MINIMUM_SUPPORT : (double) args[1];
+            int minimumSetSize = args[2] == null ? DEFAULT_MINIMUM_SET_SIZE : (int) args[2];
+            int size = args[3] == null ? DEFAULT_SIZE : (int) args[3];
+
+            return new FrequentItemSetsAggregationBuilder(context, fields, minimumSupport, minimumSetSize, size);
+        }
+    );
+
+    static {
+        ContextParser<Void, MultiValuesSourceFieldConfig.Builder> metricParser = MultiValuesSourceFieldConfig.parserBuilder(
+            false, // scriptable
+            true,  // timezone aware
+            false, // filtered
+            true   // format
+        );
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(), (p, n) -> metricParser.parse(p, null).build(), FIELDS);
+        PARSER.declareDouble(ConstructingObjectParser.optionalConstructorArg(), MINIMUM_SUPPORT);
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), MINIMUM_SET_SIZE);
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), Aggregation.CommonFields.SIZE);
+    }
+
+    private final List<MultiValuesSourceFieldConfig> fields;
+    private final double minimumSupport;
+    private final int minimumSetSize;
+    private final int size;
+
+    public FrequentItemSetsAggregationBuilder(
+        String name,
+        List<MultiValuesSourceFieldConfig> fields,
+        double minimumSupport,
+        int minimumSetSize,
+        int size
+    ) {
+        super(name);
+        this.fields = fields;
+        if (minimumSupport <= 0 || minimumSupport > 1) {
```

Comment:
```suggestion
        if (minimumSupport <= 0.0 || minimumSupport > 1.0) {
```

This is a suggestion for readability.  When I initially saw this line I thought it must be checking integers as the values being compared against were integers.